workers/unity/Assets/Fps/Scripts/GameLogic/Session/TrackPlayerSystem.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections.Generic;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.Commands;
+using Improbable.Gdk.Movement;
+using Improbable.Gdk.Session;
+using Improbable.PlayerLifecycle;
+using Improbable.Worker.CInterop;
+using Improbable.Worker.CInterop.Query;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Fps
+{
+    [DisableAutoCreation]
+    public class TrackPlayerSystem : ComponentSystem
+    {
+        private bool sentPlayerStateRequest;
+        private long? sentPlayerStateQueryId;
+        private ComponentGroup sessionGroup;
+        private CommandSystem commandSystem;
+        private ILogDispatcher logDispatcher;
+
+        private readonly EntityQuery playerStateQuery = new EntityQuery
+        {
+            Constraint = new ComponentConstraint(PlayerState.ComponentId),
+            ResultType = new SnapshotResultType()
```

Comment:
I believe this returns all components on matching entities. If we only want to return `PlayerState`, set `ResultType` to `new SnapshotResultType(new List<uint> { PlayerState.ComponentId })`